content/bn/docs/concepts/architecture/garbage-collection.md

```diff
@@ -0,0 +1,136 @@
+
+{{< note >}}
+ক্রস-নেমস্পেস মালিকানা রেফারেন্স ডিজাইন দ্বারা নিষিদ্ধ। নেমস্পেস নির্ভরশীলরা ক্লাস্টার-স্কোপড অথবা নেমস্পেসড মালিকদের নির্দিষ্ট করতে পারে। একটি নেমস্পেসড মালিক **অবশ্যই** নির্ভরশীলের সাথে একই নেমস্পেসে থাকতে হবে। যদি এটি না থাকে, তবে মালিকানা রেফারেন্সটি অনুপস্থিত হিসেবে বিবেচিত হয়, এবং মালিকদের সমস্ত অবস্থিতি যাচাই করা হলে নির্ভরশীলটি মুছে ফেলা হবে। 
+
+ক্লাস্টার-স্কোপড নির্ভরশীলরা শুধুমাত্র ক্লাস্টার-স্কোপড মালিকদের উল্লেখ করতে পারে। v1.20+ এ, যদি একটি ক্লাস্টার-স্কোপড নির্ভরশীল একটি নেমস্পেসড প্রকারকে মালিক হিসেবে নির্দিষ্ট করে, তবে এটি একটি অযোগ্য মালিকানা রেফারেন্স হিসেবে বিবেচিত হয়, এবং এটি গার্বেজ কালেক্ট করা যাবে না।
+
+v1.20+ এ, যদি গার্বেজ কালেক্টর একটি অবৈধ ক্রস-নেমস্পেস `ownerReference` সনাক্ত করে, অথবা একটি ক্লাস্টার-স্কোপড নির্ভরশীল একটি নেমস্পেসড প্রকারের মালিককে রেফারেন্স করে, তবে একটি সতর্কতা ইভেন্ট যার কারণে `OwnerRefInvalidNamespace` এবং একটি `involvedObject` থাকে যা অবৈধ নির্ভরশীল হিসেবে রিপোর্ট হয়। এধরনের ইভেন্ট দেখতে আপনি চালাতে পারেন kubectl get events `kubectl get events -A --field-selector=reason=OwnerRefInvalidNamespace`।
+{{< /note >}}
+
+## ক্যাসকেডিং ডিলিশন {#cascading-deletion}
+
+Kubernetes  সেসব অবজেক্ট পরীক্ষা করে এবং মুছে ফেলে, যেগুলোর আর মালিকানা রেফারেন্স নেই, যেমন `ReplicaSet` মুছে ফেললে বাকি থাকা পডগুলো। যখন আপনি একটি অবজেক্ট মুছে ফেলেন, তখন আপনি নিয়ন্ত্রণ করতে পারেন যে কুবার্নেটিস ঐ অবজেক্টের ডিপেনডেন্টসগুলো স্বয়ংক্রিয়ভাবে মুছে ফেলবে কিনা, এই প্রক্রিয়াকে বলা হয় *কাসকেডিং ডিলিশন*। কাসকেডিং ডিলিশনের দুটি ধরনের থাকে, যা নিচে ব্যাখ্যা করা হয়েছে:
+
+* ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন
+* ব্যাকগ্রাউন্ড ক্যাসকেডিং ডিলিশন
+
+আপনি গার্বেজ কালেকশন কিভাবে এবং কখন মালিকানা রেফারেন্সসমূহের মাধ্যমে রিসোর্স মুছে ফেলে Kubernetes {{<glossary_tooltip text="Finalizers" term_id="finalizer">}} ব্যবহার করে নিয়ন্ত্রণ করতে পারেন।
+
+
+### ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন {#foreground-deletion}
+
+ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশনে, আপনি যে মালিকানা অবজেক্টটি মুছে ফেলছেন সেটি প্রথমে একটি *ডিলিশন ইন প্রগ্রেস* অবস্থায় চলে যায়। এই অবস্থায়, মালিকানা অবজেক্টের সাথে নিম্নলিখিত ঘটনা ঘটে:
+
+* Kubernetes API সার্ভার অবজেক্টের `metadata.deletionTimestamp` ক্ষেত্রটিকে সেই সময়ে সেট করে যখন অবজেক্টটি মুছে ফেলার জন্য চিহ্নিত করা হয়।
+* Kubernetes API সার্ভার এছাড়াও `metadata.finalizers` ক্ষেত্রটিকে `foregroundDeletion` এ সেট করে।
+* অবজেক্টটি Kubernetes API-এর মাধ্যমে দৃশ্যমান থাকে যতক্ষণ না মুছে ফেলার প্রক্রিয়া সম্পূর্ণ হয়।
+
+মালিকানা অবজেক্টটি ডিলিশন ইন প্রগ্রেস অবস্থায় চলে গেলে, কন্ট্রোলার নির্ভরশীল অবজেক্টগুলি মুছে ফেলে। সমস্ত নির্ভরশীল অবজেক্ট মুছে ফেলার পর কন্ট্রোলার মালিকানা অবজেক্টটি মুছে ফেলবে। এই মুহূর্তে, অবজেক্টটি Kubernetes API-এ আর দৃশ্যমান থাকে না।
+
+ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন চলাকালীন, একমাত্র নির্ভরশীলরা যারা মালিক মুছে ফেলার প্রতিবন্ধকতা সৃষ্টি করে তা হল যাদের `ownerReference.blockOwnerDeletion=true` ক্ষেত্র রয়েছে। আরো জানার জন্য [ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করুন।](/bn/docs/tasks/administer-cluster/use-cascading-deletion/#use-foreground-cascading-deletion)
```

Comment:
Use the English version as reference `[ফরগ্রাউন্ড ক্যাসকেডিং ডিলিশন ব্যবহার করুন](/docs/tasks/administer-cluster/use-cascading-deletion/#use-foreground-cascading-deletion)।`